packages/flow-client/src/events.ts

```diff
@@ -1,6 +1,10 @@
-import type { publish, subscribe } from '../models';
+import type { ToolResult, AddInput, publish, subscribe } from '../models';
 
-export type FlowClientOutgoingMessage = Exclude<publish, string>;
+// Internal client-sent message types used by FlowClient
+export type FlowClientOutgoingMessagePrivate = Exclude<publish, string>;
+
+// client-sent message types controlled by the user
+export type FlowClientOutgoingMessage = AddInput | ToolResult;
```

Comment:
Did you consider possibly adding dedicated methods for these messages, something like `addInput()` and `sendToolResult()` or something, rather than creating this public/private type split?